docs/logging.md

```diff
@@ -0,0 +1,137 @@
+# Feature Logging
+
+To extract feature scores of a feature set or a model the `ltr_log` search extension.
+The `ltr_log` extension is able to capture feature values of a `sltr` query use inside the main search query when
+`named` or used inside a `rescore_query`.
+
+Example to output the feature scores for the feature set `my_feature_set` and for the model `my_model`:
+
+```json
+{
+  "query" : {
+    "bool" : {
+      "must" : {
+        "sltr" : {
+          "params" : {
+            "query_string" : "a search query"
+          },
+          "featureset": "my_feature_set",
+          "_name": "logged_featureset"
+        },
+        "filter": {
+          "ids" : {
+            "type" : "my_type",
+            "values" : ["1", "4"]
+          }
+        }
+      }
+    }
+  },
+  "rescore": [
+    {
+      "window_size": 3,
+      "query": {
+        "rescore_query": {
+          "sltr": {
+            "params" : {
+              "query_string" : "a search query"
+            },
+            "model": "my_model"
+          }
+        }
+      }
+    }
+  ],
+  "size": 3,
+  "ext": {
+    "ltr_log" : {
+      "log_specs": [
+        {
+          "name": "log_entry1",
+          "named_query": "logged_featureset",
```

Comment:
this example doesn't seem to work, because the query isn't named. Additionally I tried to add a "_name" property to the sltr query and that gets parsed, but the ltr_log isn't able to find it:

```
{
    "query": {
        "match": { "all": "main" }
    },
    "_source": ["title"],
    "rescore": [
        {
            "window_size": 10,
            "query": {
                "rescore_query": {
                    "sltr": {
                        "_name": "test_named_sltr",
                        "params": {
                            "query_string": "main"
                        },
                        "model": "enwiki_32153k_500t"
                    }
                }
            }
        }
    ],
    "size": 5,
    "ext": {
        "ltr_log": {
            "log_specs": [
                {
                    "name": "log_entry1",
                    "named_query": "test_named_sltr",
                    "missing_as_zero": true
                }
            ]
        }
    }
}
```

```
{
  "took": 46,
  "timed_out": false,
  "_shards": {
    "total": 4,
    "successful": 1,
    "failed": 3,
    "failures": [
      {
        "shard": 1,
        "index": "enwiki_content_first",
        "node": "ZuuouRxJQKahDb8zPTqEcw",
        "reason": {
          "type": "illegal_argument_exception",
          "reason": "No query named [test_named_sltr] found"
        }
      }
    ]
  },
  "hits": {
    "total": 781,
    "max_score": -1.7289679,
    "hits": []
  }
}
```